### ISSUE TITLE
Ability to clear quick unlock

### ISSUE THREAD

**Issue by sengsational** [kalaspuffar/secure-quick-reliable-login#320]
*Add a 'three dot' menu item to clear quick unlock*

There is now no indication in the app that the SQRL client is in the quick unlock enabled state.  I think that it would be good to add a menu item in the 'three dot' menu that indicated the state and allowed the user to wipe the quick unlock state.

In every other app in the universe, this menu item would be called "log out".  If I didn't think the techie community would freak out and blow a gasket,  I'd even like to call it "log out"!  And I'd like the menu item to change to "log in" if the quick unlock was not enabled, and allow the user to just type the full SQRL code, and set the quick unlock state without going to a web site first. 

I realize that I might be alone in my opinion here, so we can discuss.

**Comment by sesam** [comment]
I basically like the idea of keeping with what people are used to. There are other password apps out there. KeePass calls it quick unlock. There are "password wallets" that are kept open or unlocked. Doors have day- and night regimes, might be called half locked and fully locked. An identity could be in readyness or locked down.
Or from another perspective, the risk with quick unlock is that cryo- or other kinds of instant RAM dumps from the phone would expose a quickunlock-able identity, while after locking it down ot is not, ideally, anywhere in RAM anymore. Maybe using quickunlock all the time or never could be a per ID setting. The app could even look at domain hostnames and suggest to disable quickunlock for an identity you used with domains containing "bank" or "exchange" or ending in ".gov/mil/onion" :)
From this perspective though, I'd want to also have a pick identity #1 for these important domains, and #2 with quickunlock always on for these other domains. The idea being, that for i.e. banking I'm ok with typing a long password once a month, and for chats and forums I'd prefer a short pass check but much more often, and type the long pass for the lower security identity only after rebooting and say after I have been away or optionally with a warning after new hardware devices have been added or removed. I realize this two-current identities approach isn't currently supported and surely it's too early now to decide if that is a good sqrl useage approach.

**Comment by alexhauser** [comment]
Hi @sengsational ,

> I think that it would be good to add a menu item in the 'three dot' menu that indicated the state and allowed the user to wipe the quick unlock state.

While I agree that for power users like us it would be nice to know whether the QuickPass is active and to be able to disable it, I think we should absolutely keep everything as simple as possible for the "normals" that will use the app once SQRL gains traction. I'd rather have 10 power users complain about the missing feature x or y than having 1000 "normal" users being overwhelmed by all the bells and whistles that we've put in.

So in my opinion, such an feature would need to be optional and disabled by default, or burried somewhere deep in the guts of the app. 

> In every other app in the universe, this menu item would be called "log out". If I didn't think the techie community would freak out and blow a gasket, I'd even like to call it "log out"!

Gasket blown ;-)

Jokes aside, I think calling it "log out" would confuse the heck out of people. The app is for logging in to websites. But then the options for "log in" and "log out" would do something completely different? I think that would be asking for trouble. "_Clear QuickPass_" for me sounds just about right.

Cheers, Alex

**Comment by alexhauser** [comment]
Hi @sesam ,

> Maybe using quickunlock all the time or never could be a per ID setting.

I think that's already the way it is. In issue #317, Daniel wrote:
>Settings are saved on the identity and are identity specific...

I gernerally like the idea of taking this a step further, but this is clearly also a power user feature, which should not get in the way of non tech-savvy people's daily use of the app. So I agree with your final assessment that it's probably too early for adding those kinds of finesses.

Best regards, Alex

**Comment by kalaspuffar** [comment]
Hi everyone.

I think this is an interesting discussion and feature request. I've read a lot from people that want a rekey feature for handling the old site logins and so on and all these features are good to keep in mind for when SQRL has been released and maybe can be a feature that will keep this application a good choice for the Android platform over other applications.

I think we could change the name of the advanced settings in the dot menu to "Options" for instance and then in the options menu, we could have Advanced settings or options where you could activate or clear the quick pass for biometrics for instance. And if you activate it you could log in with your biometrics until you clear it again.

And in this menu, we could have the setting to save an encrypted site list in order to go through and change your identity data for people that want to use this feature. But I think it's an advanced user feature.

We still have reported bugs in the application so focusing on closing the issues we have here on Github or that have been reported in the forums should be focus no 1. And then we can keep discussing these features and keep discussions open here until we feel we have a good grasp on how these features should work and if they will benefit the community.

Best regards
Daniel

**Comment by alexhauser** [comment]
Another idea for how displaying the QuickPass status could look like:

![show_quickpass](https://user-images.githubusercontent.com/4005543/58381703-b940f080-7fc0-11e9-99cf-90050b3e1d76.JPG)

Obviously, this would not be visible by default, but only once it is enabled in settings.
What do you guys think?

**Comment by sengsational** [comment]
I'm thinking something simpler looking.  Maybe a pictorial/icon thing would work.  Then we wouldn't need to agree on a word or set of words :)  

Most other applications would show a picture of a padlock that was either closed or opened, but that's gasket blowing material again because we all know QuickPass is not "unlocked".  I can't think of a pair of icons that shows "locked" and "partially unlocked", but I'm sure we could figure something out.  So then tapping the icon in the non-QuickPass state would let them type in their SQRL identity and the icon would change to the other of the pair.  Tapping again would clear QuickPass and the icon would change back.

I'm not sure having time remaining on QuickPass is all that helpful because the way I see people using this app, they're going to set the timeout to the maximum value and rarely type their entire SQRL master password.  IMO, our app should make the maximum value about a week or so, by default, and that way they users will be force-reminded of their SQRL password periodically, but have the convenience of only typing the shortened password.

When it comes down to it, if I were using my device in public, having the device in QuickPass mode is actually MORE secure.  Right?  If someone has me on video or is watching intently, they can see me type my entire SQRL password, that's an horrific leak.  If they get video of me typing the first four, no big deal.  So that's the reason I think the default should be about 5 days.

By the way, the app I have (a week or so old maybe) consistently loses it's QuickPass state, even though I have idle timeout set to 26 days.  I think it's those instance variables in the SQRL object.  I've learned (the hard way), that zero is the right number of instance variables in Android :)

**Comment by alexhauser** [comment]
Hi @sengsational ,

I completely agree with everyhting you stated above. Let's try to find a simple pictorial solution.

> By the way, the app I have (a week or so old maybe) consistently loses it's QuickPass state, even though I have idle timeout set to 26 days. I think it's those instance variables in the SQRL object. I've learned (the hard way), that zero is the right number of instance variables in Android :)

We'll definitely need to fix this. This will massively annoy people. I suggest to open a separate issue for that, shall we?

**Comment by sengsational** [comment]
Since we can clear QuickPass from the shortcut, should this be closed, or do we want to add something inside the app to clear QuickPass?

**Comment by kalaspuffar** [comment]
Hi @sengsational, @sesam and @alexhauser 

I vote for closing it.
